scrapy/utils/defer.py

```diff
@@ -104,3 +109,21 @@ def iter_errback(iterable, errback, *a, **kw):
             break
         except Exception:
             errback(failure.Failure(), *a, **kw)
+
+
+def isfuture(o):
+    # workaround for Python before 3.5.3 not having asyncio.isfuture
+    if hasattr(asyncio, 'isfuture'):
+        return asyncio.isfuture(o)
+    return isinstance(o, asyncio.futures.Future)
+
+
+def deferred_from_coro(o):
+    """Converts a coroutine into a Deferred, or returns the object as is if it isn't a coroutine"""
+    if isinstance(o, defer.Deferred):
+        return o
+    if asyncio.iscoroutine(o) or isfuture(o) or inspect.isawaitable(o):
```

Comment:
This means any `async def` function will be converted to an asyncio Task (which then is wrapped into an asyncio Future and a Twisted Deferred) and thus will use an asyncio loop. All `async def` function pass `asyncio.iscoroutine` and I don't know yet if it's possible to distinguish simpler functions from ones really using an asyncio loop and requiring the asyncio reactor. For the simpler functions returning `defer.ensureDeferred(o)` here should be enough.